File: Subarray with given sum/Subarray with given sum.py
class Solution:
    def subArraySum(self,arr, n, s): 
       #Write your code here
        if s > sum(arr):
            return [-1]
            
        i, j, total = 0, 0, 0
        
        while j < n:
            total += arr[j]
            while total > s and i<j:   
                total -= arr[i]
                i += 1
            if total == s:
                return [i+1, j+1]   
            j += 1
        return [-1]

File: Subarray with given sum/Subarray with given sum.cpp
#include <iostream>
using namespace std;
class Solution
{
    public:
    //Function to find a continuous sub-array which adds up to a given number.
    vector<int> subarraySum(vector<int>arr, int n, long long s){
        int left=0,right=0;
        
        while(right<n){
            s-=arr[right];
            while(s<0 and left<right){
                s+=arr[left++];
            }
            if(s==0)return {left+1,right+1};
            right++;
        }
        
        return {-1};
    }
};